### ISSUE TITLE
Carla crash when refreshing the canva while zooming in/out

### ISSUE THREAD

**Issue by terencode** [falkTX/Carla#836]
Refreshing the view while scrolling at the same time makes carla crash.

Backtrace:
```
#0  0x00007fffebae4249 in  () at /usr/lib/libQt5Widgets.so.5
#1  0x00007fffebae31fa in  () at /usr/lib/libQt5Widgets.so.5
#2  0x00007fffebae31a6 in  () at /usr/lib/libQt5Widgets.so.5
#3  0x00007fffebae31fa in  () at /usr/lib/libQt5Widgets.so.5
#4  0x00007fffebae3313 in  () at /usr/lib/libQt5Widgets.so.5
#5  0x00007fffebae6612 in  () at /usr/lib/libQt5Widgets.so.5
#6  0x00007fffebae68f7 in  () at /usr/lib/libQt5Widgets.so.5
#7  0x00007fffebad93cb in  () at /usr/lib/libQt5Widgets.so.5
#8  0x00007fffebafd36e in QGraphicsView::paintEvent(QPaintEvent*) () at /usr/lib/libQt5Widgets.so.5
#9  0x00007fffebef1234 in  () at /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#10 0x00007fffeb7f9b68 in QWidget::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#11 0x00007fffeb8a136f in QFrame::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#12 0x00007fffebafbd1c in QGraphicsView::viewportEvent(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#13 0x00007fffebef2f94 in  () at /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#14 0x00007ffff6d31bab in QCoreApplicationPrivate::sendThroughObjectEventFilters(QObject*, QEvent*) ()
    at /usr/lib/libQt5Core.so.5
#15 0x00007fffeb7b8e14 in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
    at /usr/lib/libQt5Widgets.so.5
#16 0x00007fffeb7c06e1 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#17 0x00007fffebe1fb3f in  () at /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#18 0x00007ffff6d31e99 in QCoreApplication::notifyInternal2(QObject*, QEvent*) ()
    at /usr/lib/libQt5Core.so.5
#19 0x00007fffeb7f2447 in QWidgetPrivate::sendPaintEvent(QRegion const&) ()
    at /usr/lib/libQt5Widgets.so.5
#20 0x00007fffeb7f2d36 in QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, int, QPainter*, QWidgetBackingStore*) () at /usr/lib/libQt5Widgets.so.5
#21 0x00007fffeb7c8db9 in  () at /usr/lib/libQt5Widgets.so.5
#22 0x00007fffeb7c8fd9 in  () at /usr/lib/libQt5Widgets.so.5
#23 0x00007fffeb7e1497 in QWidgetPrivate::syncBackingStore() () at /usr/lib/libQt5Widgets.so.5
#24 0x00007fffeb7f9ca9 in QWidget::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#25 0x00007fffeb907015 in QMainWindow::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#26 0x00007fffebf54864 in  () at /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#27 0x00007fffeb7b8e24 in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
    at /usr/lib/libQt5Widgets.so.5
#28 0x00007fffeb7c06e1 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#29 0x00007fffebe1fb3f in  () at /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#30 0x00007ffff6d31e99 in QCoreApplication::notifyInternal2(QObject*, QEvent*) ()
    at /usr/lib/libQt5Core.so.5
#31 0x00007ffff6d34f38 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) ()
    at /usr/lib/libQt5Core.so.5
#32 0x00007fffebacb896 in  () at /usr/lib/libQt5Widgets.so.5
#33 0x00007fffebad0d59 in  () at /usr/lib/libQt5Widgets.so.5
#34 0x00007ffff6d5d152 in QObject::event(QEvent*) () at /usr/lib/libQt5Core.so.5
#35 0x00007fffebadd2fc in QGraphicsScene::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#36 0x00007fffebedd254 in  () at /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#37 0x00007fffeb7b8e24 in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
    at /usr/lib/libQt5Widgets.so.5
#38 0x00007fffeb7c06e1 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#39 0x00007fffebe1fb3f in  () at /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#40 0x00007ffff6d31e99 in QCoreApplication::notifyInternal2(QObject*, QEvent*) ()
    at /usr/lib/libQt5Core.so.5
#41 0x00007ffff6d34f38 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) ()
    at /usr/lib/libQt5Core.so.5
#42 0x00007ffff6d88714 in  () at /usr/lib/libQt5Core.so.5
```

---
ArchLinux
Carla 1:2.1.alpha1.winvst.r39.g3006738b-1
Qt 5.12.1

**Comment by falkTX** [comment]
Interesting, I guess related to the other crash.
Does this happen on master?

**Comment by falkTX** [comment]
and btw, do you have the experimental "fancy eye-candy" option on?

**Comment by terencode** [comment]
This also happens on master, all the experimental canvas options are disabled.

**Comment by eeickmeyer** [comment]
Could not reproduce on Ubuntu Disco using GNOME 3.30.

**Comment by terencode** [comment]
It might have to do with recent Qt version.

**Comment by terencode** [comment]
I can still reproduce on 2.1 and Qt 5.14.2:

```
#0  0x00007ffff235cc20 in QGraphicsItem::parentItem() const () at /usr/lib/libQt5Widgets.so.5
#1  0x00007ffff235cc44 in QGraphicsItem::topLevelItem() const () at /usr/lib/libQt5Widgets.so.5
#2  0x00007ffff23a526c in  () at /usr/lib/libQt5Widgets.so.5
#3  0x00007ffff23a410e in  () at /usr/lib/libQt5Widgets.so.5
#4  0x00007ffff23a410e in  () at /usr/lib/libQt5Widgets.so.5
#5  0x00007ffff23a40ba in  () at /usr/lib/libQt5Widgets.so.5
#6  0x00007ffff23a4195 in  () at /usr/lib/libQt5Widgets.so.5
#7  0x00007ffff23a7752 in  () at /usr/lib/libQt5Widgets.so.5
#8  0x00007ffff23a79f8 in  () at /usr/lib/libQt5Widgets.so.5
#9  0x00007ffff23999a4 in  () at /usr/lib/libQt5Widgets.so.5
#10 0x00007ffff23be17b in QGraphicsView::paintEvent(QPaintEvent*) () at /usr/lib/libQt5Widgets.so.5
#11 0x00007ffff287b514 in  () at /usr/lib/python3.8/site-packages/PyQt5/QtWidgets.abi3.so
#12 0x00007ffff20972ee in QWidget::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#13 0x00007ffff2148cbf in QFrame::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#14 0x00007ffff23bcb23 in QGraphicsView::viewportEvent(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#15 0x00007ffff287d284 in  () at /usr/lib/python3.8/site-packages/PyQt5/QtWidgets.abi3.so
#16 0x00007ffff6ce2203 in QCoreApplicationPrivate::sendThroughObjectEventFilters(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#17 0x00007ffff2053341 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#18 0x00007ffff205c829 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#19 0x00007ffff296688f in  () at /usr/lib/python3.8/site-packages/PyQt5/QtWidgets.abi3.so
#20 0x00007ffff6ce24f2 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#21 0x00007ffff208f5b8 in QWidgetPrivate::sendPaintEvent(QRegion const&) () at /usr/lib/libQt5Widgets.so.5
#22 0x00007ffff208fe1c in QWidgetPrivate::drawWidget(QPaintDevice*, QRegion const&, QPoint const&, QFlags<QWidgetPrivate::DrawWidgetFlag>, QPainter*, QWidgetRepaintManager*) () at /usr/lib/libQt5Widgets.so.5
#23 0x00007ffff2066533 in  () at /usr/lib/libQt5Widgets.so.5
#24 0x00007ffff2066fe4 in  () at /usr/lib/libQt5Widgets.so.5
#25 0x00007ffff2090d27 in QWidgetPrivate::syncBackingStore() () at /usr/lib/libQt5Widgets.so.5
#26 0x00007ffff2097d14 in QWidget::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#27 0x00007ffff21b1435 in QMainWindow::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#28 0x00007ffff2828264 in  () at /usr/lib/python3.8/site-packages/PyQt5/QtWidgets.abi3.so
#29 0x00007ffff2053352 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#30 0x00007ffff205c829 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#31 0x00007ffff296688f in  () at /usr/lib/python3.8/site-packages/PyQt5/QtWidgets.abi3.so
#32 0x00007ffff6ce24f2 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#33 0x00007ffff6ce4d56 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () at /usr/lib/libQt5Core.so.5
#34 0x00007ffff2389e2f in  () at /usr/lib/libQt5Widgets.so.5
#35 0x00007ffff238fd8d in  () at /usr/lib/libQt5Widgets.so.5
#36 0x00007ffff6d0fada in QObject::event(QEvent*) () at /usr/lib/libQt5Core.so.5
#37 0x00007ffff239e4ac in QGraphicsScene::event(QEvent*) () at /usr/lib/libQt5Widgets.so.5
#38 0x00007ffff288c7f4 in  () at /usr/lib/python3.8/site-packages/PyQt5/QtWidgets.abi3.so
#39 0x00007ffff2053352 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#40 0x00007ffff205c829 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#41 0x00007ffff296688f in  () at /usr/lib/python3.8/site-packages/PyQt5/QtWidgets.abi3.so
#42 0x00007ffff6ce24f2 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#43 0x00007ffff6ce4d56 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () at /usr/lib/libQt5Core.so.5
#44 0x00007ffff6d3b244 in  () at /usr/lib/libQt5Core.so.5
#45 0x00007ffff60a36be in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#46 0x00007ffff60a5531 in  () at /usr/lib/libglib-2.0.so.0
#47 0x00007ffff60a5571 in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#48 0x00007ffff6d3a890 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/libQt5Core.so.5
#49 0x00007ffff6ce105c in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/libQt5Core.so.5
#50 0x00007ffff6ce9066 in QCoreApplication::exec() () at /usr/lib/libQt5Core.so.5
#51 0x00007ffff296531f in  () at /usr/lib/python3.8/site-packages/PyQt5/QtWidgets.abi3.so
#52 0x00007ffff7b2b09f in PyCFunction_Call () at /usr/lib/libpython3.8.so.1.0
#53 0x00007ffff7b1e3d2 in _PyObject_MakeTpCall () at /usr/lib/libpython3.8.so.1.0
#54 0x00007ffff7bdbc51 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.8.so.1.0
#55 0x00007ffff7bc8a9d in _PyFunction_Vectorcall () at /usr/lib/libpython3.8.so.1.0
#56 0x00007ffff7bd758e in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.8.so.1.0
--Type <RET> for more, q to quit, c to continue without paging--
#57 0x00007ffff7bc78f4 in _PyEval_EvalCodeWithName () at /usr/lib/libpython3.8.so.1.0
#58 0x00007ffff7c4ed73 in PyEval_EvalCode () at /usr/lib/libpython3.8.so.1.0
#59 0x00007ffff7c4edc8 in  () at /usr/lib/libpython3.8.so.1.0
#60 0x00007ffff7c53063 in  () at /usr/lib/libpython3.8.so.1.0
#61 0x00007ffff7aeddf0 in PyRun_FileExFlags () at /usr/lib/libpython3.8.so.1.0
#62 0x00007ffff7af7aa4 in PyRun_SimpleFileExFlags () at /usr/lib/libpython3.8.so.1.0
#63 0x00007ffff7c5f81e in Py_RunMain () at /usr/lib/libpython3.8.so.1.0
#64 0x00007ffff7c5f909 in Py_BytesMain () at /usr/lib/libpython3.8.so.1.0
#65 0x00007ffff7dcf023 in __libc_start_main () at /usr/lib/libc.so.6
#66 0x000055555555505e in _start ()
```

**Comment by falkTX** [comment]
If this happens often, make sure to disable the experimental options.
The eye-candy can trigger this.

It is not easy to debug at all right now, I will only handle these kinda crashes after the frontend move to c++ is completed.

**Comment by terencode** [comment]
This is 100% reproducible for me and all eye-candy options are disabled.

**Comment by falkTX** [comment]
I wonder if there is an easy way to reproduce that matches your setup...
what exactly do you do to trigger this?

I am thinking of scripting this, xdotool is able to trigger commands, might work..

**Comment by terencode** [comment]
I spam CTRL+R at the same time I scroll up/down.

**Comment by falkTX** [comment]
damn that was easy :)
thanks!

also I see now I get a crash with a simple refresh while being zoomed in.
wtf this did not used to crash so much!
I will try to fix this...

**Comment by falkTX** [comment]
ok I know why it crashes so much, or rather, where - the qt drop shadow effect with blur.
when the effect has no blur, there is no crash.

I do like the shadow effect though... maybe I can reimplement and fix it in a custom class.

**Comment by falkTX** [comment]
too much work to do it myself ;(
oh well, better stability than tiny fanciness

So, this should be fixed in 4a1c2630e12c13e50dcbf38014085f75ff8bfb93
Please confirm. and if yes, close the ticket.

**Comment by terencode** [comment]
Ok, well fixed indeed.
